Update all dependencies

papeg opened the pull request:
googletest: release-1.11.0
hlslib: v1.4.1
clang: 12.0.1
OpenCL: 2.0
IntelSDK: 20.4.0

papeg committed:
Update all dependencies

googletest: release-1.11.0
hlslib: v1.4.1
clang: 12.0.1
OpenCL: 2.0
IntelSDK: 20.4.0

papeg commented:
Hello, we updated SMI so we can better integrate it into our project. We would be glad if you can review this. Best Regards

TizianoDeMatteis commented on CMakeLists.txt:
```diff
@@ -106,21 +107,18 @@ function(smi_target TARGET_NAME CONNECTION_FILE HOST_SOURCE KERNELS NUM_RANKS)
         set(FPGA_REPORT_TARGET ${TARGET_NAME}_${KERNEL_NAME}_aoc_report)
         add_custom_target(${FPGA_REPORT_TARGET}
-                COMMAND ${IntelFPGAOpenCL_AOC} ${AOC_COMMAND} ${FPGA_SRC_FILES} -rtl -report
+                COMMAND ${IntelFPGAOpenCL_AOC} ${KERNEL_GENERATED_PATH} -I${KERNEL_BIN_DIR} ${AOC_COMMAND} -rtl -report
```
Why is this change necessary?

papeg replied:
The new compiler version is not able to compile two cl files together, so we switched to compiling only the kernel file and then include the smi_generated_device.cl from there.

TizianoDeMatteis commented on CMakeLists.txt:
```diff
@@ -200,7 +196,7 @@ endfunction()
 function(fpga_target TARGET_NAME HOST_SOURCE KERNEL GENERATE_KERNEL)
 
-    set(FPGA_SOURCES)               # list of transformed user device files (one per program)
+    set(PGA_SOURCES)               # list of transformed user device files (one per program)
```
typo

papeg replied:
oops. gonna fix this

TizianoDeMatteis commented on codegen/templates/host.cl:
```diff
@@ -104,7 +104,7 @@ SMI_Comm SmiInit_{{ name }}(
     for (int i = num_kernels - 1; i >= 0; i--)
     {
-        queues[i].enqueueTask(kernels[i]);
+        queues[i].enqueueNDRangeKernel(kernels[i], cl::NullRange, cl::NDRange(1));
```
Is there any advantage in using `eneuqueNDRangeKernel`?

papeg replied:
enqueueTask is no more available in cl2

TizianoDeMatteis commented on examples/host/gesummv_onchip.cpp:
```diff
@@ -15,6 +15,11 @@
 #include <utils/utils.hpp>
 #define TILE_SIZE 128 //define this as used in the opencl kernel
 
+#if !defined(CL_CHANNEL_1_INTELFPGA)
```
I think that this should not be necessary

TizianoDeMatteis commented on examples/kernels/gesummv_rank0.cl:
```diff
@@ -14,6 +14,7 @@
 
 */
 
+#include "smi_generated_device.cl"
```
These include (and the ones in the successive file) should not be necessary with the old CMakeList

TizianoDeMatteis reviewed (changes requested):
Hello,
thanks for your PR!

I've left some comments, in particular the changes to the CMake file are not clear to me.

Have you double-checked that tests still work? I was not able to reproduce them (but maybe something is wrong with my configuration)

papeg commented:
Thanks for the fast review. Currently i am not able to verify the tests on the master branch

papeg committed:
fix typo

papeg committed:
use correct env flag for num of emulate devices

papeg committed:
add support for the new emulation vendor string

the new intel SDK introduced an vendor string only for the emulation device which is not supported in hlslib.
because the vendor string is also used in initOpenCL directly i introduced it into SMI,
but in perspective it should be integrated into hlsblib and used from there
